fastembed/embedding.py

```diff
@@ -1,694 +1,24 @@
-import functools
-import json
-import os
-import shutil
-import tarfile
-import tempfile
-from abc import ABC, abstractmethod
-from itertools import islice
-from multiprocessing import get_all_start_methods
-from pathlib import Path
-from typing import Any, Callable, Dict, Generator, Iterable, List, Optional, Tuple, Union
+from typing import Optional
 
-import numpy as np
-import onnxruntime as ort
-import requests
-from tokenizers import AddedToken, Tokenizer
-from tqdm import tqdm
-from huggingface_hub import snapshot_download
-from huggingface_hub.utils import RepositoryNotFoundError
 from loguru import logger
 
-from fastembed.parallel_processor import ParallelWorkerPool, Worker
+from fastembed.text.text_embedding import TextEmbedding
 
+logger.warning(
+    "DefaultEmbedding, FlagEmbedding, JinaEmbedding are deprecated."
+    " Use TextEmbedding instead."
+)
 
-def iter_batch(iterable: Union[Iterable, Generator], size: int) -> Iterable:
-    """
-    >>> list(iter_batch([1,2,3,4,5], 3))
-    [[1, 2, 3], [4, 5]]
-    """
-    source_iter = iter(iterable)
-    while source_iter:
-        b = list(islice(source_iter, size))
-        if len(b) == 0:
-            break
-        yield b
+DefaultEmbedding = TextEmbedding
+FlagEmbedding = TextEmbedding
```

Comment:
I think this retains enough backward compatibility for most users.

Reply:
that was our thought as well